acme/dns_challenge_cloudflare.go

```diff
@@ -26,26 +29,35 @@ func NewDNSProviderCloudFlare(cloudflareEmail, cloudflareKey string) (*DNSProvid
 		}
 	}
 
-	c := &DNSProviderCloudFlare{
-		client: cloudflare.New(&cloudflare.Options{Email: cloudflareEmail, Key: cloudflareKey}),
-		ctx:    context.Background(),
-	}
-
-	return c, nil
+	return &DNSProviderCloudFlare{
+		authEmail: cloudflareEmail,
+		authKey:   cloudflareKey,
+	}, nil
 }
 
 // Present creates a TXT record to fulfil the dns-01 challenge
 func (c *DNSProviderCloudFlare) Present(domain, token, keyAuth string) error {
-	fqdn, value, ttl := DNS01Record(domain, keyAuth)
+	fqdn, value, _ := DNS01Record(domain, keyAuth)
 	zoneID, err := c.getHostedZoneID(fqdn)
 	if err != nil {
 		return err
 	}
 
-	record := newTxtRecord(zoneID, fqdn, value, ttl)
-	err = c.client.Records.Create(c.ctx, record)
+	rec := cloudFlareRecord{
+		Type:    "TXT",
+		Name:    unFqdn(fqdn),
+		Content: value,
+		TTL:     120,
```

Comment:
That's the minimum valid TTL CloudFlare allows. So this was meant as a precaution in case `DNS01Record` will be changed to return a lower value anytime in the future. If you want i can bring back the clamp function from the current CloudFlare provider: https://github.com/xenolf/lego/blob/master/acme/dns_challenge_cloudflare.go#L127-L136